app/helpers/session_timeout_warning_helper.rb

```diff
@@ -37,6 +38,47 @@ def auto_session_expired_js
   def time_left_in_session
     distance_of_time_in_words(warning)
   end
+
+  def modal
+    if user_fully_authenticated?
+      FullySignedInModal.new(time_left_in_session)
+    else
+      PartiallySignedInModal.new(time_left_in_session)
+    end
+  end
+
+  FullySignedInModal = Struct.new(:time_left_in_session) do
+    include ActionView::Helpers::TagHelper
+    include ActionView::Helpers::TranslationHelper
+
+    def message
+      t('notices.timeout_warning.signed_in.message_html',
+        time_left_in_session: content_tag(:span, time_left_in_session, id: 'countdown'))
+    end
+
+    def continue
+      t('notices.timeout_warning.signed_in.continue')
+    end
+
+    def sign_out
+      t('notices.timeout_warning.signed_in.sign_out')
+    end
+  end
+
+  class PartiallySignedInModal < FullySignedInModal
```

Comment:
it looks like these classes are Duck types (need to conform to same API) -- perhaps we should test that they do? Or have them inherit from another class that raises if they do not implement one of these methods? (`sign_out` `continue` and `message` )